modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/MonocleWindowManager.java

```diff
@@ -121,7 +121,7 @@ boolean maximizeWindow(MonocleWindow window) {
 
     boolean requestFocus(MonocleWindow window) {
         int index = getWindowIndex(window);
-        if (index != -1) {
+        if (index != -1 && window.isVisible()) {
```

Comment:
This change is sufficient for the test to pass, are there any other scenarios for which the changes in Window.java are needed. If so could you please add the relevant tests.

Reply:
You are correct. The change happened, because of the leak for first noticeable in the Window class. I've now removed the change from the Window class because it's not required.